src/Core/Grid/Definition/Factory/OutstandingGridDefinitionFactory.php

```diff
@@ -0,0 +1,267 @@
+
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Outstanding', [], 'Admin.Navigation.Menu');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        return (new ColumnCollection())
+            ->add(
+                (new DataColumn('id_invoice'))
+                    ->setName($this->trans('ID', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'id_invoice',
+                    ])
+            )
+            ->add(
+                (new DataColumn('date_add'))
+                    ->setName($this->trans('Date', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'date_add',
+                    ])
+            )
+            ->add(
+                (new DataColumn('customer'))
+                    ->setName($this->trans('Customer', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'customer',
+                    ])
+            )
+            ->add(
+                (new DataColumn('company'))
+                    ->setName($this->trans('Company', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'company',
+                    ])
+            )
+            ->add(
+                (new BadgeColumn('risk'))
+                    ->setName($this->trans('Risk', [], 'Admin.Orderscustomers.Feature'))
+                    ->setOptions([
+                        'field' => 'risk',
+                        'color_field' => 'color',
+                        'badge_type' => '',
+                    ])
+            )
+            ->add(
+                (new DataColumn('outstanding_allow_amount'))
+                    ->setName($this->trans('Outstanding allowance', [], 'Admin.Orderscustomers.Feature'))
+                    ->setOptions([
+                        'field' => 'outstanding_allow_amount',
+                    ])
+            )
+            ->add(
+                (new DataColumn('outstanding'))
+                    ->setName($this->trans('Current outstanding', [], 'Admin.Orderscustomers.Feature'))
+                    ->setOptions([
+                        'field' => 'outstanding',
+                    ])
+            )
+            ->add(
+                (new ActionColumn('actions'))
+                    ->setName($this->trans('Actions', [], 'Admin.Global'))
+                    ->setOptions([
+                        'actions' => $this->getRowActions(),
+                    ])
+            );
+    }
+
+    protected function getFilters()
+    {
+        $risks = [];
+        foreach (Risk::getRisks(Context::getContext()->language->id) as $risk) {
```

Comment:
At least the context must be in the constructor.
If possible, even `Risk::getRisks`

Reply:
@ks129 Comment not applied.